node/lib/util/commit.js

```diff
@@ -1067,73 +1086,110 @@ exports.amendMetaRepo = co.wrap(function *(repo,
             }
             const subIndex = yield subRepo.index();
             yield stageFiles(subRepo, staged, subIndex);
-            subCommits[subName] = yield exports.amendRepo(subRepo, message);
+            subCommits[subName] = yield exports.amendRepo(subRepo, subMessage);
             return;                                                   // RETURN
         }
 
         const commit = yield commitRepo(subRepo,
                                         repoStatus.staged,
                                         all,
-                                        message,
+                                        subMessage,
                                         false,
                                         signature);
         if (null !== commit) {
             subCommits[subName] = commit.tostrS();
         }
     }));
 
-    const index = yield repo.index();
-    yield stageOpenSubmodules(index, subs);
-    yield stageFiles(repo, status.staged, index);
+    let metaCommit = null;
+    if (null !== message) {
 
-    const metaCommit = yield exports.amendRepo(repo, message);
+        const index = yield repo.index();
+        yield stageOpenSubmodules(index, subs);
+        yield stageFiles(repo, status.staged, index);
+        metaCommit = yield exports.amendRepo(repo, message);
+    }
     return {
         meta: metaCommit,
         subs: subCommits,
     };
 });
 
 /**
+ * Return a string describing the signature of an amend commit, to be used in
+ * an editor prompt, based on the specified `currentSignature` that is the
+ * current signature used in a normal commit, and the specified `lastSignature`
+ * used on the commit to be amended.
+ *
+ * @param {NodeGit.Signature} currentSignature
+ * @param {NodeGit.Signature} lastSignature
+ * @return {String}
+ */
+exports.formatAmendSignature = function (currentSignature, lastSignature) {
+    assert.instanceOf(currentSignature, NodeGit.Signature);
+    assert.instanceOf(lastSignature, NodeGit.Signature);
+
+    let result = "";
+    if (lastSignature.name() !== currentSignature.name() ||
+        lastSignature.email() !== currentSignature.email()) {
+        result += `\
+Author:    ${lastSignature.name()} <${lastSignature.email()}>
+`;
+    }
+    const time = lastSignature.when();
+    const offset = time.offset();
+    const prefix = offset < 0 ? "-" : "";
+    const date = new Date(time.time() * 1000);
+
+    // TODO: do something user-locale-aware
+
+    const formatted = new Intl.DateTimeFormat("en-US", {
+        hour12: false,
+        year: "numeric",
+        month: "numeric",
+        day: "numeric",
+        hour: "numeric",
+        minute: "numeric",
+        second: "numeric",
+    }).format(date);
+    result += `\
+Date:      ${formatted} ${prefix}0${Math.abs(time.offset())}00
```

Comment:
I just (a) read the source code and tests and (b) tested manually.  It is minutes.